### PR TITLE
Pal 653 eureka beans

### PR DESCRIPTION
**Pull request by developer01189998819991197253**

### CODE REVIEW

**Review by m78233 — DISMISSED** [review]

File: charts/create-kafka-queues/values.yaml [hunk 1]
```diff
@@ -16,30 +16,10 @@ replicaCount: 1
...
-- name: policy
-  replicationFactor: 1
-  partitions: 1
-- name: query-scope
-  replicationFactor: 1
-  partitions: 1
-- name: results-controller
-  replicationFactor: 1
-  partitions: 1
-
```
Comment (dev930018) [inline charts/create-kafka-queues/values.yaml]: These will need to be re-added here, no doubt removed because of the effects of `-P pi`

Comment (dev930018) [reply charts/create-kafka-queues/values.yaml]: Make sure present in both the charts/create-kafka-queues and the top-level maven module create-kafka-queues

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java [hunk 2]
```diff
@@ -59,18 +54,13 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ApplicationConfiguration.class);
 
     /**
-     * A generic resolver from service names to {@link URI}s
-     * Uses Eureka if available, otherwise uses the Spring yaml configuration value directly as a URI (useful for k8s)
+     * Simple data service simple data service.
```
Comment (dev930018) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java]: Reword this a bit??

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java [hunk 3]
```diff
@@ -15,40 +15,43 @@
...
 import uk.gov.gchq.palisade.service.Service;
 import uk.gov.gchq.palisade.service.data.request.AuditRequest;
 import uk.gov.gchq.palisade.service.data.web.AuditClient;
 
-import java.net.URI;
-import java.util.concurrent.Executor;
-import java.util.function.Supplier;
-
+/**
+ * The type Audit service which implements {@link Service}
```
Comment (dev930018) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java]: A lot of these could be better...

File: data-service/src/main/resources/application-eureka.yaml [hunk 4]
```diff
@@ -33,16 +33,17 @@ eureka:
...
   client:
     serviceUrl:
       defaultZone: http://${eureka.instance.hostname}:${eureka.instance.port}/eureka/
     enabled: true
     fetchRegistry: true
     registry-fetch-interval-seconds: 5
 
-# Eureka discovery by lookup for registered service by name
-# i.e. map from what-i-call-a-service -> what-that-service-calls-itself
+# Discovery by spring.application.name
```
Comment (dev930018) [inline data-service/src/main/resources/application-eureka.yaml]: Is this still valid documentation considering the empty-strings?

Probably better to say "explicitly provide default urls to feign in order to use discovery by name"

File: palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/web/AuditClient.java [hunk 5]
```diff
@@ -23,15 +23,25 @@
 
 import uk.gov.gchq.palisade.service.palisade.request.AuditRequest;
 
-import java.net.URI;
-
-@FeignClient(name = "audit-service", url = "undefined")
+/**
+ * The interface Audit client which uses Feign to resolve services called audit-service or falls back to values set in the relevant profiles yaml.
```
Comment (dev930018) [inline palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/web/AuditClient.java]: Reword?
Uses service urls if provided, otherwise discovery by name with eureka

File: services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/service/ManagedService.java [hunk 6]
```diff
@@ -36,6 +36,9 @@
     private final ManagedClient managedClient;
     private final Supplier<Collection<URI>> uriSupplier;
 
+    public static final int VALID_RESPONSE = 200;
+    public static final int ERROR_RESPONSE = 404;
```
Comment (dev930018) [inline services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/service/ManagedService.java]: While it's out-of-scope, these should strictly come from the HttpStatus object

**Review by dev930018 — CHANGES_REQUESTED** [review]
Lots of changes that could be made to documentation, but may be out-of-scope

File: data-service/src/main/resources/application-eureka.yaml (hunk 4 above)
Comment (developer01189998819991197253) [reply data-service/src/main/resources/application-eureka.yaml]: changed

File: charts/create-kafka-queues/values.yaml (hunk 1 above)
Comment (developer01189998819991197253) [reply charts/create-kafka-queues/values.yaml]: readded

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java (hunk 2 above)
Comment (developer01189998819991197253) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java]: reworded

File: data-service/src/main/resources/application-eureka.yaml (hunk 4 above)
Comment (developer01189998819991197253) [reply data-service/src/main/resources/application-eureka.yaml]: reworded

File: palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/web/AuditClient.java (hunk 5 above)
Comment (developer01189998819991197253) [reply palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/web/AuditClient.java]: reworded

File: services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/service/ManagedService.java (hunk 6 above)
Comment (developer01189998819991197253) [reply services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/service/ManagedService.java]: quickly changed

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java (hunk 3 above)
Comment (developer01189998819991197253) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java]: fixed

**Review by dev-958 — DISMISSED** [review]

File: charts/data-service/values.yaml [hunk 7]
```diff
@@ -31,12 +31,6 @@ resources:
     cpu: 250m
     memory: 512Mi
 
-autoscaling:
-  targetCPUUtilizationPercentage: 50
-  targetMemoryUtilizationPercentage: 50
-  minReplicas: 1
-  maxReplicas: 3
-
```
Comment (ac74475) [inline charts/data-service/values.yaml]: Is this removal intended?

File: palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/web/AuditClient.java [hunk 8]
```diff
@@ -23,15 +23,25 @@
...
 
+    /**
+     * Gets health.
+     *
+     * @return the health
+     */
     @GetMapping(path = "/actuator/health", produces = "application/json")
-    Response getHealth(final URI url);
-
+    Response getHealth();
```
Comment (ac74475) [inline palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/web/AuditClient.java]: all the other services and clients have removed the getHealth method, why not here?

File: resource-service/src/test/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceServiceTest.java [hunk 9]
```diff
@@ -43,12 +43,11 @@
...
 public class SimpleResourceServiceTest {
-    private final ClientConfiguration clientConfiguration = Mockito.mock(ClientConfiguration.class);
-    private final SimpleResourceService service = new SimpleResourceService(clientConfiguration);
+    private final DiscoveryClient discoveryClient = Mockito.mock(DiscoveryClient.class);
+    private final SimpleResourceService service = new SimpleResourceService("data-service");
 
     @Before
     public void setUp() throws URISyntaxException {
-        Mockito.when(clientConfiguration.getClientUri(Mockito.anyString())).thenReturn(Optional.of(new URI("http://data-service-uri")));
     }
```
Comment (ac74475) [inline resource-service/src/test/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceServiceTest.java]: remove method since it is otherwise empty

File: user-service/src/main/resources/application.yaml [hunk 10]
```diff
@@ -38,15 +37,6 @@ management:
...
 
-feign:
-  client:
-    config:
-      default:
-        connectTimeout: 5000
-        readTimeout: 5000
-        loggerLevel: basic
-    default-config: default
-
```
Comment (ac74475) [inline user-service/src/main/resources/application.yaml]: This section still exists in most of the other service application.yaml files. Are you sure it should be removed here and not in the other application.yaml files?

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java [hunk 11]
```diff
@@ -15,40 +15,43 @@
...
 import uk.gov.gchq.palisade.service.data.request.AuditRequest;
 import uk.gov.gchq.palisade.service.data.web.AuditClient;
 
-import java.net.URI;
-import java.util.concurrent.Executor;
-import java.util.function.Supplier;
-
+/**
+ * Wraps a feign client to provide an implementation of {@link Service}
+ */
```
Comment (nw1984) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java:27]: Is this true  still - I see no mention of a feign client in the definition.
Suggest something along the lines of implements AuditService providing an implementation of {@link Service}

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/PalisadeService.java [hunk 12]
```diff
@@ -24,35 +23,44 @@
...
 
-import java.net.URI;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.Executor;
-import java.util.function.Supplier;
 
+/**
+ * Wraps a feign client to provide an implementation of {@link Service}
+ */
```
Comment (nw1984) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/PalisadeService.java:32]: Ditto.

**Review by ac74475 — CHANGES_REQUESTED** [review]
Most of the Javadoc comments that have been added in this PR are pointless in their current state as they don't provide any extra information than what you can get from the method signature. The Javadoc should provide the contextual information to help someone new use the method. Not just there to stop SonarQube complaining.

**Review by nw1984 — DISMISSED** [review]

File: resource-service/src/main/resources/application-k8s.yaml [hunk 13]
```diff
@@ -24,8 +24,6 @@ spring:
 spring:
   profiles: k8s-config
 
-# Kubernetes discovery by DNS for the pod name
-# i.e. map from what-i-call-a-service -> name-of-that-service's-pod
```
Comment (pd104923) [inline resource-service/src/main/resources/application-k8s.yaml]: Is there a reason that the comment `# Kubernetes discovery via DNS to the service loadbalancer` was not added to this file?

File: services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/config/TaskConfiguration.java [hunk 14]
```diff
@@ -29,15 +29,30 @@
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
+/**
+ * The type Task configuration.
+ */
```
Comment (pd104923) [inline services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/config/TaskConfiguration.java:36]: The JavaDoc comments in this TaskConfiguration class do not explain the methods properly.

File: resource-service/src/main/resources/application-static.yaml [hunk 15]
```diff
@@ -28,4 +28,4 @@ spring:
 # i.e. map from what-i-call-a-service -> service-URI
 web:
   client:
-    data-service: http://localhost:8082
+    data-service: "http://localhost:8082"
```
Comment (pd104923) [inline resource-service/src/main/resources/application-static.yaml]: Do this value really need to be inside quotes?

**Review by pd104923 — DISMISSED** [review]
A couple of questions but also agree about the JavaDoc comments not being descriptive enough

File: charts/data-service/values.yaml (hunk 7 above)
Comment (developer01189998819991197253) [reply charts/data-service/values.yaml]: no, for some reason updating the image tag removed some values from values.yamls, i have re-added them

File: palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/web/AuditClient.java (hunk 8 above)
Comment (developer01189998819991197253) [reply palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/web/AuditClient.java]: removed

File: resource-service/src/test/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceServiceTest.java (hunk 9 above)
Comment (developer01189998819991197253) [reply resource-service/src/test/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceServiceTest.java]: removed

File: user-service/src/main/resources/application.yaml (hunk 10 above)
Comment (developer01189998819991197253) [reply user-service/src/main/resources/application.yaml]: readded

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java (hunk 11 above)
Comment (developer01189998819991197253) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java:27]: updated

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/PalisadeService.java (hunk 12 above)
Comment (developer01189998819991197253) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/PalisadeService.java:32]: updated

File: resource-service/src/main/resources/application-k8s.yaml (hunk 13 above)
Comment (developer01189998819991197253) [reply resource-service/src/main/resources/application-k8s.yaml]: reverted and readded

File: resource-service/src/main/resources/application-static.yaml (hunk 15 above)
Comment (developer01189998819991197253) [reply resource-service/src/main/resources/application-static.yaml]: removed quotes to be a bit more consistent

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java [hunk 16]
```diff
@@ -59,18 +55,14 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ApplicationConfiguration.class);
 
     /**
-     * A generic resolver from service names to {@link URI}s
-     * Uses Eureka if available, otherwise uses the Spring yaml configuration value directly as a URI (useful for k8s)
+     * Simple data service bean created with instances of auditService, palisadeService and dataReader
+     * which is a simple implementation of {@link DataService} used for examples and demos
```
Comment (ac74475) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java]:
```suggestion
     * Simple data service bean created with instances of auditService, palisadeService and dataReader
     * which is used by a simple implementation of {@link DataService} to connect to different data storage technologies and deserialise the data

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java [hunk 17]
```diff
@@ -83,20 +75,28 @@ public DataReader hadoopDataReader() throws IOException {
         return new HadoopDataReader();
```
Comment (ac74475) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java:82]: No javadoc on the hadoop data reader

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java [hunk 18]
```diff
@@ -15,40 +15,43 @@
...
+     * @param auditClient the audit client rest interface for the Audit Service
+     */
+    public AuditService(final AuditClient auditClient) {
         this.client = auditClient;
-        this.uriSupplier = uriSupplier;
-        this.executor = executor;
     }
 
+    /**
+     * Audit boolean which calls the Audit Client and via feign sends audit requests to the Audit service
```
Comment (ac74475) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/AuditService.java]:
```suggestion
     * Calls the Audit Client and via feign sends audit requests to the Audit service returning TRUE if the audit message was received successfully. 
     * Audit service implementations may have different definitions of 'received successfully'.
```

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/PalisadeService.java [hunk 19]
```diff
@@ -24,35 +23,45 @@
...
+     */
+    public PalisadeService(final PalisadeClient palisadeClient, final Executor executor) {
         this.client = palisadeClient;
-        this.uriSupplier = uriSupplier;
         this.executor = executor;
     }
 
+    /**
+     * Gets data request config which calls the PalisadeClient and via feign returns a CompletableFuture of DataRequestConfigs by request.
+     * the GetDataRequestConfig method in the Palisade Service extracts resources from request and check they are a subset of the original RegisterDataRequest resources
```
Comment (ac74475) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/PalisadeService.java]:
```suggestion
     * Gets a @{link DataRequestConfig} by calling the PalisadeClient via feign. It then validates that the resources from the request are a subset of the original RegisterDataRequest resources
```

**Review by ac74475 — CHANGES_REQUESTED** [review]
a few minor changes.

File: services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/config/TaskConfiguration.java (hunk 14 above)
Comment (developer01189998819991197253) [reply services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/config/TaskConfiguration.java:36]: added

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java (hunk 17 above)
Comment (developer01189998819991197253) [reply data-service/src/main/java/uk/gov/gchq/palisade/service/data/config/ApplicationConfiguration.java:82]: added

**Review by dev-958 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Review by m78233 — APPROVED** [review]

**Review by dev930018 — APPROVED** [review]
